hardware/GoodweAPI.cpp

```diff
@@ -166,11 +166,12 @@ int GoodweAPI::hash(std::string str)
 
 	long hash = 5381;
 	int i = 0;
-	int c;
+	int c = str[i++];
 	
-	while(c = str[i++])
+	while(c)
 	{
 		hash = ((hash << 5) + hash) + c;
+		c = str[i++];
 	}
```

Comment:
I still find this a bit strange way todo this
You know the length of the string ( str.size() )
You could use

while (i<str.size())

or even

for (i=0; i<str.size(); i++)

Both look a bit better